File: android/app/src/main/java/app/tourism/data/dto/place/ReviewDto.kt
package app.tourism.data.dto.place

import app.tourism.domain.models.details.Review

data class ReviewDto(
    val id: Long,
    val mark_id: Long,
    val images: List<String>,
    val message: String,
    val points: Int,
    val created_at: String,
    val user: UserDto
) {
    fun toReview() = Review(
        id = id,
        placeId = mark_id,
        rating = points,
        user = user.toUser(),
        date = created_at,
        comment = message,
        picsUrls = images
    )
}

File: iphone/Maps/Tourism/Data/Network/DTO/Details/ReviewDTO.swift
import Foundation

struct ReviewDTO: Codable {
  let id: Int64
  let markId: Int64
  let images: [String]
  let message: String
  let points: Int
  let createdAt: String
  let user: UserDTO
  
  func toReview() -> Review {
    return Review(
      id: id,
      placeId: markId,
      rating: points,
      user: user.toUser(),
      date: createdAt,
      comment: message,
      picsUrls: images
    )
  }
}
